File: solutions/go/0089_格雷编码.go
package main

// 89. 格雷编码
// 经典公式：gray(i) = i ^ (i >> 1)。
func grayCode(n int) []int {
	size := 1 << n
	result := make([]int, size)
	for i := 0; i < size; i++ {
		result[i] = i ^ (i >> 1)
	}
	return result
}

File: solutions/java/0089_格雷编码.java
public class Solution {
    /**
     * 89. 格雷编码
     * 经典公式：gray(i) = i ^ (i >> 1)。
     */
    public List<Integer> grayCode(int n) {
        List<Integer> result = new ArrayList<>();
        int size = 1 << n;
        for (int i = 0; i < size; i++) {
            result.add(i ^ (i >> 1));
        }
        return result;
    }
}